src/pj_internal.c

```diff
@@ -134,36 +136,193 @@ size_t pj_strlcpy(char *dst, const char *src, size_t siz) {
  * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
  * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
 
-  Source: http://www.i-pi.com/Training/EthicalHacking/Solutions/strlcpy.c
+  Source: http://cvsweb.openbsd.org/cgi-bin/cvsweb/src/lib/libc/string/strlcpy.c
 
-********************************************************************/
-    register char *d = dst;
-    register const char *s = src;
-    register size_t n = siz;
+***************************************************************************************/
+    const char *osrc = src;
+    size_t nleft = dsize;
 
-    /* Copy as many bytes as will fit */
-    if (n != 0 && --n != 0) {
-        do {
-            if ((*d++ = *s++) == 0)
+    /* Copy as many bytes as will fit. */
+    if (nleft != 0) {
+        while (--nleft != 0) {
+            if ((*dst++ = *src++) == '\0')
                 break;
-        } while (--n != 0);
+        }
     }
 
-    /* Not enough room in dst, add NUL and traverse rest of src */
-    if (n == 0) {
-        if (siz != 0)
-            *d = '\0';      /* NUL-terminate dst */
-        while (*s++)
+    /* Not enough room in dst, add NUL and traverse rest of src. */
+    if (nleft == 0) {
+        if (dsize != 0)
+            *dst = '\0';		/* NUL-terminate dst */
+        while (*src++)
             ;
     }
 
-    return(s - src - 1);    /* count does not include NUL */
+    return(src - osrc - 1);	/* count does not include NUL */
```

Comment:
so would mine - making return look like a function call seems utterly strange to me. But this is code lifted directly from the OpenBSD repo, and I think I will just reimplement using strncpy and friends,  also in order to avoid the inline license and attribution stuff.